src/System.Management.Automation/engine/parser/Compiler.cs

```diff
@@ -3074,8 +3076,9 @@ public object VisitPipeline(PipelineAst pipelineAst)
                 Expression invokePipe = Expression.Call(
                     CachedReflectionInfo.PipelineOps_InvokePipeline,
                     input.Cast(typeof(object)),
-                    firstCommandExpr != null ? ExpressionCache.FalseConstant : ExpressionCache.TrueConstant,
-                    Expression.NewArrayInit(typeof(CommandParameterInternal[]), pipelineExprs),
+                    firstCommandExpr != null && !pipelineAst.BackgroundProcess ? ExpressionCache.FalseConstant : ExpressionCache.TrueConstant,
+                    Expression.NewArrayInit(typeof(CommandParameterInternal[]), 
+                        !pipelineAst.BackgroundProcess ? pipelineExprs : Utils.EmptyArray<Expression>()),
                     Expression.Constant(pipeElementAsts),
                     redirectionExpr,
```

Comment:
I don't see any tests for redirections. I assume the redirections are also done in the background, and hence should not be generated in the foreground.